### PR TITLE
devicetree: miscellaneous improvements

### PR DESCRIPTION
**Pull request by mbolivar-nordic**
- allow path type properties in inferred bindings
- allow iterating over property values with a new helper, and use it to improve the documentation for inferred bindings
- overhaul the bindings documentation

I was working on something else and ended up separating these into a standalone prep work PR.

### CODE REVIEW

File: doc/guides/dts/intro.rst [hunk 1]
```diff
@@ -339,6 +339,107 @@ interrupts
...
+     - How to write
+     - Example
+
+   * - string
+     - Double quoted
+     - ``a-string = "hello, world!";``
+
+   * - int
+     - between angle brackets (``<`` and ``>``)
+     - ``an-int = "<1>";``
```
Comment (gmarull) [inline doc/guides/dts/intro.rst:369]: nit: `an-int = <1>;`

File: doc/guides/dts/intro.rst [hunk 2]
```diff
@@ -339,6 +339,107 @@ interrupts
...
+- 64-bit integers are written as two 32-bit cells in big-endian order. The value
+  0xaaaa0000bbbb1111 would be written ``<0xaaaa0000 0xbbbb1111>``.
+
+- The ``a-byte-array`` property value is the three bytes 0x00, 0x01, and 0xab, in
+  that order.
+
+- Parentheses, arithmetic operators, and bitwise operators are allowed. The
+  ``bar`` property contains a single cell with value 64::
+
+  	bar = <2 * (1 << 5)>;
```
Comment (gmarull) [inline doc/guides/dts/intro.rst:418]: If I'm not wrong, the whole expression needs to be enclosed in parenthesis, i.e. `bar = <(2 * (1 << 5))>`

**Review by gmarull — COMMENTED** [review]
LGTM, some minor comments

**Comment by mbolivar-nordic** [comment]
@gmarull thank you for catching those. Now fixed.

**Review by gmarull — APPROVED** [review]

**Comment by mbolivar-nordic** [comment]
Latest push includes fixes for errors I found and some other twwe.

**Comment by mbolivar-nordic** [comment]
dropped this part after discussion with @galak :

```
Add documentation now that /chosen gets inferred bindings too.
```

**Review by jfischer-no — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Pull request merged**